curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb3f62b10c336bada1c70c.md

```diff
@@ -24,13 +24,13 @@ assert.match(code, /\bcheckpoints\b\s*\.\s*forEach\s*\(/s);
 You should have a `checkpoint` parameter inside the `forEach` callback function.
 
 ```js
-assert.match(code, /\s*checkpoints\s*\.\s*forEach\s*\(\s*checkpoint\s*=>\s*{/s);
+assert.match(code, /\s*checkpoints\s*\.\s*forEach\s*\(\s*\(?checkpoint\)?\s*=>\s*{/s);
```

Comment:
```suggestion
assert.match(code, /\s*checkpoints\s*\.\s*forEach\s*\(\s*(\(\s*checkpoint\s*\)|checkpoint)\s*=>\s*{/s);
```
Doing it this way will make sure there's either both parenthesis or none.

Reply:
Hang on. My thing also works. The question mark afterwards means it is optional. Is your solution easier to read though?

Reply:
I'm a bit nitpicking, but the optional parenthesis on each side makes ie. this match:
```
checkpoints.forEach(checkpoint) => {
```